common/src/app/common/pages/migrations.cljc

```diff
@@ -291,83 +282,91 @@
+                         :fill-color-ref-file (:fill-color-ref-file shape)
+                         :fill-color-ref-id (:fill-color-ref-id shape)
+                         :fill-opacity (:fill-opacity shape)}
+                  clean-attrs (d/without-nils attrs)]
+              (cond-> shape
+                (d/not-empty? clean-attrs)
+                (assoc :fills [clean-attrs]))))
+
+          (assign-strokes [shape]
+            (let [attrs {:stroke-style (:stroke-style shape)
+                         :stroke-alignment (:stroke-alignment shape)
+                         :stroke-width (:stroke-width shape)
+                         :stroke-color (:stroke-color shape)
+                         :stroke-color-ref-id (:stroke-color-ref-id shape)
+                         :stroke-color-ref-file (:stroke-color-ref-file shape)
+                         :stroke-opacity (:stroke-opacity shape)
+                         :stroke-color-gradient (:stroke-color-gradient shape)
+                         :stroke-cap-start (:stroke-cap-start shape)
+                         :stroke-cap-end (:stroke-cap-end shape)}
+                  clean-attrs (d/without-nils attrs)]
+              (cond-> shape
+                (d/not-empty? clean-attrs)
+                (assoc :strokes [clean-attrs]))))
+
+          (update-object [object]
             (cond-> object
-              (and (not (= :text (:type object))) (nil? (:strokes object)))
-              (set-strokes)))
+              (and (not (cph/text-shape? object))
+                   (not (contains? object :strokes)))
+              (assign-strokes)
 
-          (update-page [_ page]
-            (update page :objects #(d/mapm update-object %)))]
-    (update data :pages-index #(d/mapm update-page %))))
+              (and (not (cph/text-shape? object))
+                   (not (contains? object :fills)))
+              (assign-fills)))
 
-;; Add fills and strokes to components
+          (update-container [container]
+            (update container :objects d/update-vals update-object))]
 
-(defmethod migrate 16
+    (-> data
+        (update :pages-index d/update-vals update-container)
+        (update :components d/update-vals update-container))))
+
+(defmethod migrate 17
   [data]
-  (letfn [(update-object [_ object]
+  (letfn [(update-object [object]
             (cond-> object
-              (and (not (= :text (:type object))) (nil? (:strokes object)))
-              (set-strokes)
+              ;; First: delete the old unused props from image shapes
+              (cph/image-shape? object)
+              (dissoc :fill-color :fill-opacity)
+
+              ;; Second: fix already migrated shapes (only applicable
+              ;; if you are using nightly builds)
+              (and (cph/image-shape? object)
+                   (contains? object :fills)
+                   (= 1 (count (:fills object)))
+                   (= (:fill-color object) (get-in object [:fills 0 :fill-color])))
```

Comment:
I would also check the fill-color and the opacity are the default values (1 and #B1B2B5)

Reply:
the main idea here is checking if the current :fills has the same value as prev and no more fills are added. I think we don't need more checks here, because previously we only have a single hidden fill on images, and the previous migration creates the :fills with single element.